mathesar/models/base.py

```diff
@@ -434,9 +434,13 @@ def insert_records_to_existing_table(self, existing_table, data_files, mappings=
         from_table = self._sa_table
         target_table = existing_table._sa_table
         engine = self._sa_engine
+        if mappings:
+            sa_col_mappings = [[from_col._sa_column, target_col._sa_column] for from_col, target_col in mappings]
```

Comment:
Each of these `_sa_column` attributes comes with its own SQLAlchemy `Table` instance. Just pass the names instead (see the other comment)